assets/javascripts/discourse/components/chat-live-pane.js

```diff
@@ -418,22 +443,29 @@ export default Component.extend({
         }
       });
     }
-    const lastReadId =
-      this.currentUser.chat_channel_tracking_state[this.chatChannel.id]
-        ?.chat_message_id;
+    const lastReadId = this._getLastReadId();
     if (!lastReadId) {
       return;
     }
 
     this.set("lastSendReadMessageId", lastReadId);
     const indexOfLastReadyMessage =
       this.messages.findIndex((m) => m.id === lastReadId) || 0;
-    const newestUnreadMessage = this.messages[indexOfLastReadyMessage + 1];
+    let newestUnreadScrollTarget = indexOfLastReadyMessage + 1;
+    let newestUnreadMessage = this.messages[newestUnreadScrollTarget];
 
     if (newestUnreadMessage) {
       newestUnreadMessage.set("newestMessage", true);
-      // We have the last read message from lookup, but now we need the index of the message,
-      // so that we can scroll to the message directly after it.
+      const previousMessagesToDisplay = 3;
```

Comment:
I'd be fine with reducing this to 2. To avoid the "new message" indicator being below the fold in the off chance that these 3 messages are long. (There's only so much we can do here without major goldplating, but 2 is a tiny bit less risky than 3.)